src/validators/validator.ts

```diff
@@ -0,0 +1,102 @@
+export interface Validator<A> {
+    validate: (x: A) => boolean;
+    errorMessage?: object;
+}
+
+export interface ValidatorEntry {
+    name: string,
+    options?: any;
+}
+
+export const defaultValidator: Validator<any> = {
+    validate: (_x: any) => true
+}
+
+export function combineValidators<A>(v1: Validator<A>, v2: Validator<A>): Validator<A> {
+    let combined: Validator<A>;
+    combined = {
+        validate: (x: A) => {
+            const res1: boolean = v1.validate(x);
+            const res2: boolean = v2.validate(x);
+            if (!res1) {
+                combined.errorMessage = v1.errorMessage;
+            } else if (!res2) {
+                combined.errorMessage = v2.errorMessage;
+            }
+            return res1 && res2;
+        },
+    }
+    return combined;
+}
+
+export function requiredValidator(element, type) {
+    if (element.required) {
+        switch(type) {
+            case("input"):
+            case("textarea"):
+            case("select"):
+              if (element.validator) {
+                element.validator.unshift("requiredInput");
+              } else {
+                element.validator= ["requiredInput"]
```

Comment:
How about renaming this validator to `requiredField` instead of `requiredInput` since it is shared with the textarea and the select?